tools/pre-swap-checker/contracts/PreSwapChecker.sol

```diff
@@ -175,10 +181,66 @@ contract PreSwapChecker {
     if (order.signer.token == address(wethContract)) {
       uint256 allowance = wethContract.allowance(order.sender.wallet, wrapper);
       if (allowance < order.signer.amount) {
-        errors[errorCount] = "LOW_SENDER_ALLOWANCE_ON_WRAPPER";
+        errors[errorCount] = "SENDER_WRAPPER_ALLOWANCE_LOW";
+        errorCount++;
+      }
+    }
+
+    // Check valid token registry handler for sender
+    if (hasValidKind(order.sender.kind, swap)) {
+      // Check the order sender
+      if (order.sender.wallet != address(0)) {
+        // The sender was specified
+        // Check if sender kind interface can correctly check balance
+        if (order.sender.kind == ERC721_INTERFACE_ID && !hasValidERC71Interface(order.sender.token)) {
+          errors[errorCount] = "SENDER_INVALID_ERC721";
+          errorCount++;
+        } else {
+          // Check the order sender token balance when sender is not WETH
+          if (order.sender.token != address(wethContract)) {
+            //do the balance check
+            if (!hasBalance(order.sender)) {
+              errors[errorCount] = "SENDER_BALANCE_LOW";
+              errorCount++;
+            }
+          }
+
+          // Check their approval
+          if (!isApproved(order.sender, swap)) {
+            errors[errorCount] = "SENDER_ALLOWANCE_LOW";
+            errorCount++;
+          }
+        }
+      }
+    } else {
+      errors[errorCount] = "SENDER_TOKEN_KIND_UNKNOWN";
+      errorCount++;
+    }
+
+     // Check valid token registry handler for signer
+    if (hasValidKind(order.signer.kind, swap)) {
+      // Check if sender kind interface can correctly check balance
+      if (order.signer.kind == ERC721_INTERFACE_ID && !hasValidERC71Interface(order.signer.token)) {
```

Comment:
Can't comment on the function itself. But is `hasValidERC71Interface` a valid function? What if the token just doesn't implement the function `_supportsInterface` but is an ERC721?

Reply:
_supportInterface is actually calling ERC165Checker which is an OpenZeppelin library. Its responsibility is to check that the ERC721 follows ERC165 which is required for ERC721 tokens.